File: dataset/output/OPENAI_gpt-4o/Task199_OPENAI_gpt-4o.c
#include <stdio.h>
#include <stdbool.h>

typedef struct {
    int hour;
    int minute;
} Clock;

Clock create_clock(int hour, int minute) {
    int totalMinutes = (hour * 60 + minute) % (24 * 60);
    Clock clock = {totalMinutes / 60, totalMinutes % 60};
    return clock;
}

Clock add_minutes(Clock clock, int minutes) {
    return create_clock(clock.hour, clock.minute + minutes);
}

bool are_equal(Clock clock1, Clock clock2) {
    return clock1.hour == clock2.hour && clock1.minute == clock2.minute;
}

void print_clock(Clock clock) {
    printf("%02d:%02d
", clock.hour, clock.minute);
}

int main() {
    Clock clock1 = create_clock(10, 30);
    Clock clock2 = add_minutes(clock1, 35);
    Clock clock3 = create_clock(11, 5);
    print_clock(clock2);  // 11:05
    printf("%d
", are_equal(clock2, clock3));  // 1 (True)
    return 0;
}

File: dataset/output/OPENAI_gpt-4o/Task199_OPENAI_gpt-4o.java
import java.util.Objects;

public class Clock {
    private int hour;
    private int minute;

    public Clock(int hour, int minute) {
        int totalMinutes = (hour * 60 + minute) % (24 * 60);
        this.hour = totalMinutes / 60;
        this.minute = totalMinutes % 60;
    }

    @Override
    public String toString() {
        return String.format("%02d:%02d", this.hour, this.minute);
    }

    public Clock addMinutes(int minutes) {
        return new Clock(this.hour, this.minute + minutes);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Clock clock = (Clock) o;
        return hour == clock.hour && minute == clock.minute;
    }

    @Override
    public int hashCode() {
        return Objects.hash(hour, minute);
    }

    public static void main(String[] args) {
        Clock clock1 = new Clock(10, 30);
        Clock clock2 = clock1.addMinutes(35);
        Clock clock3 = new Clock(11, 5);
        System.out.println(clock2);  // 11:05
        System.out.println(clock2.equals(clock3));  // True
    }
}